package.json

```diff
@@ -4,13 +4,13 @@
   "description": "Nodejs bindings for Secure Reliable Transport SDK",
   "main": "index.js",
   "scripts": {
-    "preinstall": "npm install git-clone && node scripts/build-srt-sdk.js",
-    "install": "node-gyp rebuild -j 8",
-    "rebuild": "node-gyp rebuild",
-    "clean": "node-gyp clean",
+    "install": "npm run build-srt && npm run rebuild",
+    "build-srt": "node scripts/build-srt-sdk.js",
+    "rebuild": "$(npm bin)/node-gyp rebuild -j $(echo \"console.log(require('os').cpus().length)\" | node)",
+    "clean": "$(npm bin)/node-gyp clean",
     "test": "$(npm bin)/jasmine",
```

Comment:
@birme One question, why are we using `npm bin` here in the first place?

I have applied it consistently in the present patch, but realized it might no be needed at all.

If for example `jasmine` is installed in the closest node_modules bin path, the shell run by npm for this script will select it over any global installed executable (in case you wanted to ensure the local version is used).

You would rather have to enforce a global version to be used, but that would only make sense if that package was installed in the local project bin path at all.

Reply:
I want to use the local node_modules version to not have to "pollute" the global node_modules space. For example `jasmine` is a dev-dependency

Reply:
Yes. And that means it gets installed in your `node_modules`(of the project). With that the case, as I tried to explain, the npm script shell will run the local bin without needing an explicit path (your *local* `node_modules/.bin` is prepended to the `PATH` var of that created shell in which the package script runs).

In short -> What you want will exactly work without the `npm bin` thing. Using the local package installed bin (no matter if dev-dep or runtime dep) is the default over any global bin path installs :) 

So, if that is what you want and it is the only requirement here, we can remove all `$(npm bin)/` here, because the behavior you want already is provided by putting there just `jasmine` or `eslint`. npm will make sure local bins will resolve this before user/system-wide installs.

Reply:
Aha, I understand. I didn't know that.

Reply:
No worries. Sorry if my explanation was confusing at first maybe. By "local" i was meaning the node_modules created in the project root here. Obviously there is also anything installed in /usr/local/bin and other places on the system that one could call "local", which I was meaning by "global" :) 

So, yes anyhow, if you want to do something straight forward like this, dont think too complicated, those package scripts have been created exactly with that idea in mind to be able to launch the installed dependencies bins without having to specifiy a path :) npm doesn't do much really that is true, but that's one thing it definitely manages hehe :D

Reply:
It is funny that the `npm` docs are not very clear on this (but they aren't on many things): https://docs.npmjs.com/misc/scripts#path 

They say "added" to the path, which does not mean prepended, in the sense that it does not clarify order or resolution between those and other installs.

Here is a more historical and clear statement where this law of nature was introduced, `npm bin` output is prepending the shell `PATH` for lifecycle scripts since v1: https://nodejs.org/en/blog/npm/npm-1-0-global-vs-local-installation/#test-runners-and-stuff

```
npm makes ./node_modules/.bin the first entry in the PATH environment variable when running any lifecycle scripts, so this will work fine, even if your program is not globally installed:
```

Reply:
As always very educating :)
Please go ahead and remove the usage of $(npm bin) in this PR if you want to.

Reply:
I am mostly stumbling upon my own doubts as I am trying to answer these questions :D